### PR TITLE
[Feat #636] Add localization to GameMenu

### LINKED ISSUE #636: [Feature]: Apply translations/localization to the `GameMenu`
### Feature Summary

We should use `i18n` to translate the `GameMenu` 

### Detailed Description

We should pull all of the text in the `GameMenu` into the translation json files e.g. src/translations/en.json, src/translations/es.json, and src/translations/fr.json in under a key called "game.gameMenu"

Look at src/views/LoginView for an example of a page that is correctly using $i18n to localize/translate content. The user-visible strings have been removed from the code and placed into a "login" section of the translation files in src/translations.

The page then imports and consumes the useI18n() composable to grab the t() function used to render/translate text by pulling from the appropriate language's json file. This is done using a setup() block of the page like so:

```
setup() {
const { t } = useI18n();
return { t };
},
```
From there, we can grab a given string by name in the appropriate locale using {{ t('login.joinDiscord') }} since each translation json in this case has a login attribute for all the text in the login page, with a joinDiscord child attribute for the text to show in the 'Join our discord community' button.

We should do the same thing to the `GameMenu` by extracting all the text in the page into a rules section of each translation json file (e.g. src/translations/en.json) and then configuring `GameMenu.vue` with the useI18n() composable and rendering all its text with t() to translate it to the user's selected locale.

All the text for the `GameMenu` should be stored in each language file under the key "game.dialogs.GameMenu" (generally we have one top-level key for each page e.g. "game" and "login", then specific sub-keys for components in that page.

The end result of this change should be the ability to see the text of the GameMenu in your preferred locale.

### PR DESCRIPTION
**Pull request by psinha0**
## Issue number

- Resolves #636 

## Please check the following

- [X] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [X] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

## Please describe additional details for testing this change
- Added translations/localization to menu items in `GameMenu`.
- Added translations/localization to concede/stalemate description in `GameMenu`.

### CODE REVIEW

File: src/routes/game/components/GameMenu.vue [hunk 1]
```diff
@@ -17,19 +17,19 @@
       <!-- Menu -->
       <v-list id="game-menu" class="text-surface-1" bg-color="surface-2">
         <v-list-item data-cy="rules-open" @click="shownDialog = 'rules'">
-          Rules
+          {{ `${t('game.menus.gameMenu.rules')}` }}
```
Comment (Haviles04) [inline src/routes/game/components/GameMenu.vue:20]:
```suggestion
          {{  t('game.menus.gameMenu.rules')  }}
```

I don't think we need the interpolation here or any of them in the template

File: src/routes/game/components/GameMenu.vue [hunk 2]
```diff
@@ -111,12 +116,12 @@ export default {
       },
     },
     dialogTitle() {
-      return this.showConcedeDialog ? 'Concede' : 'Request Stalemate';
+      return this.showConcedeDialog ? this.$t('game.menus.gameMenu.concede') : this.$t('game.menus.gameMenu.stalemate');
```
Comment (Haviles04) [inline src/routes/game/components/GameMenu.vue:119]:
```suggestion
      return this.t( this.showConcedeDialog ? 'game.menus.gameMenu.conceded' : 'game.menus.gameMenu.stalemate')
```

File: src/routes/game/components/GameMenu.vue [hunk 3]
```diff
@@ -111,12 +116,12 @@ export default {
...
     },
     dialogText() {
       return this.showConcedeDialog
-        ? 'The game will end and your opponent will win.'
-        : 'If your opponent agrees, the game will end in a stalemate. The request will cancel if the opponent declines or either player makes a move';
+        ? this.$t('game.menus.gameMenu.concedeDialog')
+        : this.$t('game.menus.gameMenu.stalemateDialog');
```
Comment (Haviles04) [inline src/routes/game/components/GameMenu.vue:124]: `return this.t(this.showConcedeDialog ? 'game.menus.gameMenu.concedeDialog' : 'game.menus.gameMenu.stalemateDialog') `

**Review by Haviles04 — COMMENTED** [review]
Nice job! I think we can remove the interpolation in the template and just call the t() directly.

I also suggested some changes on the computed value to DRY it up

File: src/routes/game/components/GameMenu.vue (hunk 1 above)
Comment (itsalaidbacklife) [reply src/routes/game/components/GameMenu.vue:20]: Correct we only need the ` ${} string template literals when we're combining multiple translations into a single string

**Review by itsalaidbacklife — COMMENTED** [review]
This looks good. Once the comments from @Haviles04 are addressed and the merge conflicts are resolved, this will be ready to go. The merge conflicts will hopefully be simple to resolve, someone else added a menu block to the translations so your new stuff should be added to the existing `"game.menus"` block so that the new `"game.menus.gameMenu"` block is a sibling of the existing `"game.menus.spectatorListMenu"` block.

Reach out if you get stuck or have any questions

**Comment by psinha0** [comment]
Hey! I removed all interpolations (did not realize that it works like that!), and fixed all conflicts. Hopefully should be all good now!

**Pull request merged**